pkg/search/proxy/framework/plugins/cache/cache.go

```diff
@@ -13,42 +13,65 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/apiserver/pkg/endpoints/handlers"
-	"k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/rest"
 	"k8s.io/client-go/kubernetes/scheme"
 
+	"github.com/karmada-io/karmada/pkg/search/proxy/framework"
 	"github.com/karmada-io/karmada/pkg/search/proxy/store"
 )
 
-// cacheProxy caches resource from member clusters, and handle the read request(get/list/watch) for resources.
-type cacheProxy struct {
-	store             store.RESTReader
+const (
+	// We keep a big gap between in tree plugins, to allow users to insert custom plugins between them.
+	order = 1000
+)
+
+// Cache caches resource from member clusters, and handle the read request(get/list/watch) for resources.
+// For reading requests, we redirect them to cache.
+// Users call these requests to read resources in member clusters, such as pods and nodes.
+type Cache struct {
+	store             store.Store
 	restMapper        meta.RESTMapper
 	minRequestTimeout time.Duration
 }
 
-func newCacheProxy(store store.RESTReader, restMapper meta.RESTMapper, minRequestTimeout time.Duration) *cacheProxy {
-	return &cacheProxy{
-		store:             store,
-		restMapper:        restMapper,
-		minRequestTimeout: minRequestTimeout,
-	}
+var _ framework.Plugin = (*Cache)(nil)
+
+// New creates an instance of Cache
+func New(dep framework.PluginDependency) (framework.Plugin, error) {
+	return &Cache{
+		store:             dep.Store,
+		restMapper:        dep.RestMapper,
+		minRequestTimeout: dep.MinRequestTimeout,
+	}, nil
 }
 
-func (c *cacheProxy) connect(ctx context.Context, _ schema.GroupVersionResource, _ string, _ rest.Responder) (http.Handler, error) {
-	requestInfo, _ := request.RequestInfoFrom(ctx)
-	gvr := schema.GroupVersionResource{
-		Group:    requestInfo.APIGroup,
-		Version:  requestInfo.APIVersion,
-		Resource: requestInfo.Resource,
-	}
+// Order implements Plugin
+func (c *Cache) Order() int {
+	return order
+}
+
+// SupportRequest implements Plugin
+func (c *Cache) SupportRequest(request framework.ProxyRequest) bool {
+	requestInfo := request.RequestInfo
+
+	return requestInfo.IsResourceRequest &&
+		c.store.HasResource(request.GroupVersionResource) &&
+		requestInfo.Subresource == "" &&
+		(requestInfo.Verb == "get" ||
+			requestInfo.Verb == "list" ||
+			requestInfo.Verb == "watch")
```

Comment:
This logic should belong to `karmadaProxy`, and I find that the logic of the three plug-ins seems confused.

Reply:
@ikaven1024 said this is needed，you can see him comment.

Reply:
This part logic has been reviewed by proxy's author, @ikaven1024. 
I think you can pass it.

Reply:
> This logic should belong to `karmadaProxy`, and I find that the logic of the three plug-ins seems confused.

IMOP, every plugin is individual, so it's predication shall be complete.

Reply:
Ok, thanks for explain.